lib/cylc/config.py

```diff
@@ -44,30 +45,24 @@
 from cylc.print_tree import print_tree
 from cylc.taskdef import TaskDef, TaskDefError
 from cylc.task_id import TaskID
-from cylc.task_trigger import TaskTrigger
+from cylc.task_trigger import TaskTrigger, Dependency
 from cylc.wallclock import get_current_time_string
 from isodatetime.data import Calendar
 from isodatetime.parsers import DurationParser
 from parsec.OrderedDict import OrderedDictWithDefaults
 from parsec.util import replicate
 from cylc.suite_logging import OUT, ERR
-
+from cylc.task_outputs import TASK_OUTPUT_SUCCEEDED
 
 RE_SUITE_NAME_VAR = re.compile('\${?CYLC_SUITE_(REG_)?NAME}?')
 RE_TASK_NAME_VAR = re.compile('\${?CYLC_TASK_NAME}?')
 CLOCK_OFFSET_RE = re.compile(r'(' + TaskID.NAME_RE + r')(?:\(\s*(.+)\s*\))?')
 EXT_TRIGGER_RE = re.compile('(.*)\s*\(\s*(.+)\s*\)\s*')
 NUM_RUNAHEAD_SEQ_POINTS = 5  # Number of cycle points to look at per sequence.
 
-# Replace \W characters in conditional graph expressions.
-CONDITIONAL_REGEX_REPLACEMENTS = [
-    ("\[", "_leftsquarebracket_"),
-    ("\]", "_rightsquarebracket_"),
-    ("-", "_minus_"),
-    ("\^", "_caret_"),
-    (":", "_colon_"),
-    ("\+", "_plus_"),
-]
+# Message trigger offset regex(es).
+BCOMPAT_MSG_RE_C5 = re.compile(r'^(.*)\[\s*T\s*(([+-])\s*(\d+))?\s*\](.*)$')
```

Comment:
Testing for message offsets is impacting validation so we should remove this as soon as we are confident that it is no-longer needed. In the mean time can we remove the cylc5 regex?